SECP-service/src/main/java/com/visucius/secp/Chat/ChatSocketListener.java

```diff
@@ -0,0 +1,104 @@
+package com.visucius.secp.Chat;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.visucius.secp.DTO.MessageDTO;
+import com.visucius.secp.models.User;
+import com.visucius.secp.util.JsonUtil;
+import org.apache.log4j.Logger;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.WebSocketListener;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class ChatSocketListener implements WebSocketListener, IMessageReceiver {
+
+    private static final Logger log = Logger.getLogger(ChatSocketListener.class);
+
+    private Session outbound;
+    private final User user;
+    private final IMessageHandler messageHandler;
+
+    public ChatSocketListener(User user, IMessageHandler messageHandler)
+    {
+        this.user = user;
+        this.messageHandler = messageHandler;
+    }
+
+    @Override
+    public void onWebSocketBinary(byte[] bytes, int i, int i1) {
+
+    }
+
+    @Override
+    public void onWebSocketText(String value) {
+
+        log.info("Message was sent from" + user.getUsername());
+        try
+        {
+            MessageDTO message = JsonUtil.convertStringToJson(value,MessageDTO.class);
```

Comment:
lets try to verify for the type of message here. The client can send multiple type of messages e.g getPublicKeys e.t.c